common/string_utils.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2020-2022 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ *
+ * This software product is a proprietary product of Nvidia Corporation and its affiliates
+ * (the "Company") and all right, title, and interest in and to the software
+ * product, including all associated intellectual property rights, are and
+ * shall remain exclusively with the Company.
+ *
+ * This software product is governed by the End User License Agreement
+ * provided with the software product.
+ */
+#include <fstream>
+#include <sstream>
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <stdio.h>
+#include <string.h>
+#include <memory>
+#include <cctype>
+#include <algorithm>
+#include <string>
+#include "string_utils.h"
+
+namespace string_utils
+{
+    std::string& trim(std::string& s)
+    {
+        s.erase(std::remove_if(s.begin(), s.end(), ::isspace), s.end());
+        return s;
+    }
+
+    std::string trim_copy(std::string& s)
+    {
+        std::string out = s;
+
+        return trim(out);
+    }
+
+    void to_lower(std::string& s)
+    {
+        transform(s.begin(), s.end(), s.begin(), [] (unsigned char c) { return std::tolower(c);
```

Comment:
you can skip type casting since 'c' is already has type 'unsigned char' as per lambda signature
[] (unsigned char c) { return std::tolower(static_cast < unsigned char > (c));